obp-ri-transport/src/test/java/com/tesobe/obp/transport/spi/ConnectorV0Test.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) TESOBE Ltd. 2016. All rights reserved.
+ *
+ * Use of this source code is governed by a GNU AFFERO license
+ * that can be found in the LICENSE file.
+ *
+ */
+
+package com.tesobe.obp.transport.spi;
+
+import com.tesobe.obp.transport.*;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.*;
+
+import static com.github.npathai.hamcrestopt.OptionalMatchers.hasValue;
+import static com.tesobe.obp.transport.Transport.Encoding.json;
+import static com.tesobe.obp.transport.Transport.Version.legacy;
+import static com.tesobe.obp.util.MethodMatcher.returns;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.junit.Assert.assertThat;
+
+public class ConnectorV0Test
+{
+  @Before public void setup()
+  {
+    Transport.Factory factory = Transport.factory(legacy, json)
+      .orElseThrow(RuntimeException::new);
+    Receiver responder = new MockResponderV0(factory.decoder(),
+      factory.encoder());
+    final BlockingQueue<String> in = new SynchronousQueue<>();
+    final BlockingQueue<Message> out = new SynchronousQueue<>();
+
+    // sender
+    connector = factory.connector(request ->
+    {
+      out.put(request);
+
+      return in.take();
+    });
+
+    service = Executors.newSingleThreadExecutor();
+
+    // receiver
+    service.submit(new Callable<Void>()
+    {
+      @Override @SuppressWarnings({"InfiniteLoopStatement"}) public Void call()
+        throws InterruptedException
+      {
+        for(; ; )
+        {
+          in.put(responder.respond(out.take()));
+        }
+      }
+    });
+  }
+
+  @After public void shutdown()
+  {
+    service.shutdown();
+  }
+
+  @Test public void getAccount() throws Exception
```

Comment:
You lost many tests - you should have at least two tests for each getXXX : with and without valid user